graphql/exec.go

```diff
@@ -38,7 +41,7 @@ func collectFields(reqCtx *RequestContext, selSet ast.SelectionSet, satisfies []
 			if !shouldIncludeNode(sel.Directives, reqCtx.Variables) {
 				continue
 			}
-			if !instanceOf(sel.TypeCondition, satisfies) {
+			if len(satisfies) > 0 && !instanceOf(sel.TypeCondition, satisfies) {
```

Comment:
should this be an explicit nil check? is there a difference between satisfies = nil and satisfies = []? maybe [] is not useful so the distinction is not important?

Reply:
Yeah perhaps.  Currently on `next` an empty slice will collect object fields, and skip any Fragments.  Is that useful?  Maybe, I guess it does potentially break BC if anyone is using it for this use-case.

Still, a `nil` slice vs and empty slice having different behaviours seems a little fragile to me?

Reply:
:+1: